letsencrypt/auth_handler.py

```diff
@@ -175,9 +190,8 @@ def _poll_challenges(
                         chall_update[domain].remove(achall)
                 # We failed some challenges... damage control
                 else:
-                    # Right now... just assume a loss and carry on...
                     if best_effort:
-                        comp_domains.add(domain)
+                        failed_domains.add(domain)
```

Comment:
To be honest, I think changes in this file are more complicated than they need to be. I think every change in this file can be reverted, except we should log that a challenge was failed on this line. See my comments in `client.py` for how to obtain the list of domains from `authzr`.